### PR TITLE
feat(headless): add fieldsToInclude to insight search request

### PR DESCRIPTION
**Pull request by aelfaleh**

### CODE REVIEW

File: packages/headless/src/features/insight-search/insight-search-actions.ts [hunk 1]
```diff
@@ -307,6 +309,13 @@ const buildInsightSearchRequest = (
       numberOfResults: state.pagination.numberOfResults,
     }),
     ...(cq && {cq}),
+    ...(state.fields &&
+      !state.fields.fetchAllFields && {
+        fieldsToInclude: state.fields.fieldsToInclude,
+      }),
+    ...(state.didYouMean && {
+      enableDidYouMean: state.didYouMean.enableDidYouMean,
```
Comment (nathanlb) [inline packages/headless/src/features/insight-search/insight-search-actions.ts:317]: good catch

**Review by nathanlb — APPROVED** [review]
This is good but we should probably add some tests copied over from [search-request.test.ts](https://github.com/coveo/ui-kit/blob/master/packages/headless/src/features/search/search-request.test.ts) so we avoid forgetting any more.

**Review by ThibodeauJF — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/features/insight-search/insight-search-request.test.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      (await buildInsightSearchRequest(state)).request.fieldsToInclude
+    ).toEqual(expect.arrayContaining(['foo', 'bar']));
+  });
+
+  it('#insightSearchRequest.fieldsToInclude does not holds #fieldsToInclude if #fetchAllFields is active', async () => {
+    state.fields.fieldsToInclude = ['foo', 'bar'];
+    state.fields.fetchAllFields = true;
+    expect(
+      (await buildInsightSearchRequest(state)).request.fieldsToInclude
+    ).not.toBeDefined();
```
Comment (lbergeron) [inline packages/headless/src/features/insight-search/insight-search-request.test.ts:101]:
```suggestion
    ).toBeUndefined();
```

File: packages/headless/src/features/insight-search/insight-search-request.test.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  });
+
+  it('when there are no cq expressions in state, cq is undefined', async () => {
+    expect((await buildInsightSearchRequest(state)).request.cq).toBe(undefined);
```
Comment (lbergeron) [inline packages/headless/src/features/insight-search/insight-search-request.test.ts:105]:
```suggestion
    expect((await buildInsightSearchRequest(state)).request.cq).toBeUndefined();
```

**Review by lbergeron — APPROVED** [review]
It looks all good to me!

**Pull request merged**